src/Orleans/Streams/Internal/StreamSubscriptionHandleImpl.cs

```diff
@@ -121,13 +115,6 @@ public override Task<StreamSubscriptionHandle<T>> ResumeAsync(IAsyncObserver<T>
 
             await NextItem(item, currentToken);
 
-            // check again, in case the expectedToken was changed indiretly via ResumeAsync()
-            if (expectedToken != null)
```

Comment:
I don't have a repro test, and can't easily make one because we don't have a rewindable stream provider in Orleans.  The issue was tracked down in services load test from logging.

I'd initially thought this code was needed, but each new subscription creates a new handle, so if the grain updates the subscription, this handle will simply be invalidated.  Upon further reflection, if the handle is invalidated before all of the batch events have been fed to the observer, we should throw to force the redelivery if the batch.  I'll look into this.